Linker error after using "std::chrono::steady_clock::now()"

If a am using

```
#  include <chrono>
...
std::chrono::steady_clock::now()

```
with a H745ZI, C031C6 or L476RG, I get the linker error

```
ld.exe: m-none-eabi\lib\thumb\v6-m\nofp\libc_nano.a(libc_a-gettimeofdayr.o): in function _gettimeofday_r
gettimeofdayr.c*: (.text._gettimeofday_r+0xe): undefined reference to _gettimeofday
 
collect2.exe*: error: ld returned 1 exit status

Error linking for board Nucleo-64 (STMicroelectronics_Nucleo_64)
Debug build failed for project 'STM32-C031C6' 
```

Please add he missing function to the library.

Thanks
Richard

--- a/cores/arduino/wiring_time.h
+++ b/cores/arduino/wiring_time.h
@@ -22,6 +22,7 @@
 
 #include "clock.h"
 #include "dwt.h"
+#include <sys/time.h> // for struct timeval
 
 #ifdef __cplusplus
 extern "C" {
@@ -84,6 +85,27 @@ static inline void delayMicroseconds(uint32_t us)
     oldTicks = currentTicks;
   } while (nbTicks > elapsedTicks);
 #endif
+}
+
+/**
+ * \brief gives the number of seconds and microseconds since the Epoch
+ *
+ *        based on millisecond since last power on.
+ *
+ * \note  The function is declared as weak  to be overwritten  in case of other
+ *        implementations in user file (using RTC values for example).
+ *
+ * \param tv argument is a struct timeval
+ * \param tz argument is a struct timezone (unused)
+ *
+ * \return 0
+ */
+int __attribute__((weak)) _gettimeofday(struct timeval *tv, void *tz)
+{
+  (void)tz;
+  tv->tv_sec = getCurrentMillis() / 1000;
+  tv->tv_usec = getCurrentMicros() - (tv->tv_sec * 1000000);  // get remaining microseconds
+  return 0;
 }
 
 #ifdef __cplusplus

